Framework/DataHandling/test/LoadPSIMuonBinTest.h

```diff
@@ -0,0 +1,126 @@
+#ifndef MANTID_DATAHANDLING_LOADPSIMUONBINTEST_H_
+#define MANTID_DATAHANDLING_LOADPSIMUONBINTEST_H_
+
+#include <cxxtest/TestSuite.h>
+
+#include "MantidDataHandling/LoadPSIMuonBin.h"
+#include "MantidAPI/AnalysisDataService.h"
+#include "MantidAPI/FileFinder.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/Sample.h"
+#include "MantidDataObjects/Workspace2D.h"
+#include "MantidKernel/FileDescriptor.h"
+
+using namespace Mantid;
+using namespace Mantid::DataHandling;
+using namespace Mantid::API;
+using namespace Mantid::Kernel;
+
+class LoadPSIMuonBinTest : public CxxTest::TestSuite {
+
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static LoadPSIMuonBinTest *createSuite() { return new LoadPSIMuonBinTest(); }
+  static void destroySuite(LoadPSIMuonBinTest *suite) { delete suite; }
+
+  void test_Init() {
+    LoadPSIMuonBin alg;
+    TS_ASSERT_THROWS_NOTHING(alg.initialize())
+    TS_ASSERT(alg.isInitialized())
+
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty(
+        "Filename", getTestFilePath("deltat_tdc_dolly_1529.bin")));
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("OutputWorkspace", "ws"));
+  }
+
+  void test_exec() {
+    LoadPSIMuonBin alg;
+    TS_ASSERT_THROWS_NOTHING(alg.initialize());
+    TS_ASSERT(alg.isInitialized());
+
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty(
+        "Filename", getTestFilePath("deltat_tdc_dolly_1529.bin")));
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("OutputWorkspace", "ws"));
+    TS_ASSERT_THROWS_NOTHING(alg.execute());
+
```

Comment:
you can end the exec test here and call the below a second test. The the exec test only tests that the algorithm executes